src/GitRepository.cc

```diff
@@ -0,0 +1,415 @@
+
+            continue;
+        }
+
+        if (s->status & GIT_STATUS_WT_MODIFIED)
+            wstatus = "modified";
+        if (s->status & GIT_STATUS_WT_DELETED)
+            wstatus = "deleted";
+        if (s->status & GIT_STATUS_WT_RENAMED)
+            wstatus = "renamed";
+        if (s->status & GIT_STATUS_WT_TYPECHANGE)
+            wstatus = "typechange";
+
+        if (wstatus != "")
+        {
+            filestats.handling = "unstaged";
+            filestats.changes = std::string{wstatus};
+
+
+            old_path = s->index_to_workdir->old_file.path;
+            new_path = s->index_to_workdir->new_file.path;
+
+            if (old_path && new_path && strcmp(old_path, new_path))
+                filestats.path_name = gul14::cat(old_path, " -> ", new_path);
+            else
+                filestats.path_name = old_path ? std::string{old_path} : std::string{new_path};
+
+            return_array.push_back(filestats);
+
+            continue;
+        }
+
+        // list files which are staged for next commit
+        //############################################
+
+        if (s->status & GIT_STATUS_INDEX_NEW)
+            istatus = "new file";
+        if (s->status & GIT_STATUS_INDEX_MODIFIED)
+            istatus = "modified";
+        if (s->status & GIT_STATUS_INDEX_DELETED)
+            istatus = "deleted";
+        if (s->status & GIT_STATUS_INDEX_RENAMED)
+            istatus = "renamed";
+        if (s->status & GIT_STATUS_INDEX_TYPECHANGE)
+            istatus = "typechange";
+
+        if (istatus != "")
+        {
+          
+            filestats.handling = "staged";
+            filestats.changes = std::string{istatus};
+
+            old_path = s->head_to_index->old_file.path;
+            new_path = s->head_to_index->new_file.path;
+
+            if (old_path && new_path && strcmp(old_path, new_path))
+                filestats.path_name = gul14::cat(old_path, " -> ", new_path);
+            else
+                filestats.path_name = old_path ? std::string{old_path} : std::string{new_path};
+
+            return_array.push_back(filestats);
+
+            
+            continue;
+        }
+
+        // list untracked files
+        //######################
+        if (s->status == GIT_STATUS_WT_NEW)
+        {
+
+            filestats.handling = "untracked";
+            filestats.changes = "untracked";
+            filestats.path_name = std::string(s->index_to_workdir->old_file.path);
+
+            return_array.push_back(filestats);
+
+            continue;
+        }
+
+        // list ignored files
+        //####################
+        if (s->status == GIT_STATUS_IGNORED) {
+
+            filestats.handling = "ignored";
+            filestats.changes = "ignored";
+            filestats.path_name = std::string(s->index_to_workdir->old_file.path);
+
+            return_array.push_back(filestats);
+
+            continue;
+        }
+
+        delete [] old_path;
+        delete [] new_path;
```

Comment:
Aren't `old_path` and `new_path` pointers returned by `git_status_byindex`? Why call `delete []` on them if there never was a `new`?

Reply:
I define them as const char*

Reply:
Right, I forgot that I just stole the pointer